tests/TransactionStackTest.php

```diff
@@ -12,27 +12,155 @@
 namespace Raven\Tests;
 
 use PHPUnit\Framework\TestCase;
+use Raven\TransactionStack;
 
 class TransactionStackTest extends TestCase
 {
-    public function testSimple()
-    {
-        $stack = new \Raven\TransactionStack();
-        $stack->push('hello');
-        /** @noinspection PhpVoidFunctionResultUsedInspection */
-        /** @noinspection PhpUnusedLocalVariableInspection */
-        $foo = $stack->push('foo');
-        $stack->push('bar');
-        $stack->push('world');
-        $this->assertEquals($stack->peek(), 'world');
-        $this->assertEquals($stack->pop(), 'world');
-        $this->assertEquals($stack->pop('foo'), 'foo');
-        $this->assertEquals($stack->peek(), 'hello');
-        $this->assertEquals($stack->pop(), 'hello');
-        $this->assertEquals($stack->peek(), null);
+    public function testConstructor()
+    {
+        $stack = new TransactionStack(['a', 'b']);
+
+        $this->assertAttributeEquals(['a', 'b'], 'transactions', $stack);
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The $values argument must contain string values only.
+     */
+    public function testConstructorThrowsIfValuesAreNotAllStrings()
+    {
+        new TransactionStack(['a', 1]);
+    }
+
+    public function testClear()
+    {
+        $stack = new TransactionStack();
+
+        $this->assertAttributeEmpty('transactions', $stack);
```

Comment:
Maybe is better to use the public accessor methods instead of inspecting the internal state of the class

Reply:
There is no way to access the array that holds the values of the stacm from the outside as it would defeat the purpose of the stack to access values randomly and I don't think we should change this to allow it

Reply:
There is no way to access the array that holds the values of the stack  from the outside as it would defeat the purpose of the stack to access values randomly and I don't think we should change this to allow it

Reply:
There is no way to access the array that holds the values of the stack from the outside as it would defeat the purpose of the whole class to access values randomly and I don't think we should change this to allow it

Reply:
I've changed where possible the `assertAttribute*` method calls to use more appropriate functions, but I can't do this in all places as there is no way to access the array that holds the values of the stack from the outside as it would defeat the purpose of the whole class